packages/x-date-pickers-pro/src/TimeRangePicker/TimeRangePickerTabs.tsx

```diff
@@ -0,0 +1,165 @@
+  /**
+   * Toggles visibility of the tabs allowing view switching.
+   * @default `window.innerHeight < 667` for `DesktopTimeRangePicker` and `MobileTimeRangePicker`, `displayStaticWrapperAs === 'desktop'` for `StaticTimeRangePicker`
+   */
+  hidden?: boolean;
+  /**
+   * Time tab icon.
+   * @default Time
+   */
+  timeIcon?: React.ReactNode;
+  /**
+   * Override or extend the styles applied to the component.
+   */
+  classes?: Partial<TimeRangePickerTabsClasses>;
+}
+
+export interface TimeRangePickerTabsProps extends ExportedTimeRangePickerTabsProps {}
+
+const useUtilityClasses = (classes: Partial<TimeRangePickerTabsClasses> | undefined) => {
+  const slots = {
+    root: ['root'],
+    tab: ['tab'],
+  };
+
+  return composeClasses(slots, getTimeRangePickerTabsUtilityClass, classes);
+};
+
+const TimeRangePickerTabsRoot = styled(Tabs, {
+  name: 'MuiTimeRangePickerTabs',
+  slot: 'Root',
+  overridesResolver: (_, styles) => styles.root,
+})(({ theme }) => ({
+  boxShadow: `0 -1px 0 0 inset ${(theme.vars || theme).palette.divider}`,
+  '&:last-child': {
+    boxShadow: `0 1px 0 0 inset ${(theme.vars || theme).palette.divider}`,
+    [`& .${tabsClasses.indicator}`]: {
+      bottom: 'auto',
+      top: 0,
+    },
+  },
+}));
+
+const TimeRangePickerTab = styled(Tab, {
+  name: 'MuiTimeRangePickerTabs',
+  slot: 'Tab',
+  overridesResolver: (_, styles) => styles.tab,
+})(({ theme }) => ({
+  minHeight: '48px',
+  gap: theme.spacing(1),
+}));
+
+/**
+ * Demos:
+ *
+ * - [TimeRangePicker](https://mui.com/x/react-date-pickers/time-range-picker/)
+ * - [Custom slots and subcomponents](https://mui.com/x/react-date-pickers/custom-components/)
+ *
+ * API:
+ *
+ * - [TimeRangePickerTabs API](https://mui.com/x/api/date-pickers/time-range-picker-tabs/)
+ */
+const TimeRangePickerTabs = function TimeRangePickerTabs(inProps: TimeRangePickerTabsProps) {
+  const props = useThemeProps({ props: inProps, name: 'MuiTimeRangePickerTabs' });
+  const {
+    timeIcon = <TimeIcon />,
+    hidden = typeof window === 'undefined' || window.innerHeight < 667,
+    className,
+    sx,
+    classes: classesProp,
+  } = props;
+
+  const translations = usePickerTranslations();
+  const { view, setView } = usePickerContext();
+  const { rangePosition, setRangePosition } = usePickerRangePositionContext();
+  const classes = useUtilityClasses(classesProp);
+
+  const handleChange = (event: React.SyntheticEvent, value: RangePosition) => {
+    if (rangePosition !== value) {
+      setRangePosition(value);
+    }
+    if (view !== 'hours') {
+      setView('hours');
+    }
+  };
+
+  if (hidden) {
+    return null;
+  }
+
+  return (
+    <TimeRangePickerTabsRoot
+      variant="fullWidth"
+      value={rangePosition}
+      onChange={handleChange}
+      className={clsx(className, classes.root)}
+      sx={sx}
+    >
+      <TimeRangePickerTab
+        value="start"
+        iconPosition="start"
+        icon={<React.Fragment>{timeIcon}</React.Fragment>}
```

Comment:
Do we need to fragment here? Same for the end icon

Reply:
It's an easy way out of the TS error. 🙈 
The same solution is used on other Tabs components.
Otherwise, we need to define it as `ReactElement`.

I'm not against it, would you agree that going for `React.ReactElement` here makes more sense? 🤔

Reply:
If we do the same on other Tabs components, we can keep the change for later
I do think we should fix the typing eventually, added a component in the tree for "nothing" is not optimal

Reply:
Oh, I will keep the same behavior.
I'll add a comment to consider replacing it with `React.ReactElement` if possible.

P.S. I have a slight hunch tha we might have had reasons for doing it to support certain use cases, but couldn't find it based on git history. 🙈 🤷

Reply:
I've rechecked this case and noticed that only `DateTimePickerTabs` has `React.ReactNode`, whereas `DateTimeRangePickerTabs` already uses `React.ReactElement`.
I've updated the solution to `React.ReactElement`. 😉 
https://github.com/mui/mui-x/pull/9431/commits/d63dc63ff3a08388f65b804bb9330774bd7be0e3